TFEngine.py

```diff
@@ -1427,8 +1464,24 @@ def extra_fetches_callback(seq_tags, **extra_fetches_out):
     if output_per_seq_file:
       print('Write eval results per seq to %r' % output_per_seq_file, file=log.v3)
       from Util import betterRepr
+
+      # write fetches to file
       with open(output_per_seq_file, 'w') as f:
-        f.write(betterRepr(results_per_seq) + '\n')
+        def create_output_string(st):
+          """
+          Creates a colon (;) separated output string in the order given by `requested_outputs`
+          :param str st : a seq_tag
+          """
+          value_list = [results_per_seq[st][req_out] for req_out in requested_outputs]
+          value_list = [' '.join(map(str, v)) if isinstance(v, numpy.ndarray) else str(v) for v in value_list]
+          return ';'.join(value_list)
```

Comment:
Do not do sth like this. No custom format/syntax.

Reply:
Actually, what was the problem with the format before? It should be much simpler, and better, in many ways:
* The code here is simpler.
* The parsing code will be simpler.
* It's bullet proof, this cannot break.

Reply:
No, that's not true. Reading a dict should be way faster. And of course much easier.
Also, you do not have the problem that this likely will break at some point.
The space overhead should be really low. And if this is a problem, maybe you can just use on-the-fly gzip or so.